src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -250,6 +254,20 @@ namespace System.Diagnostics.Tracing
     /// </remarks>
     public partial class EventSource : IDisposable
     {
+
+#if FEATURE_EVENTSOURCE_XPLAT
+        private static XplatEventLogger persistent_Listener = null;
+
+        static EventSource()
+        {
+            if ((persistent_Listener == null) && XplatEventLogger.IsEventSourceLoggingEnabled())
+            {
+                persistent_Listener = new XplatEventLogger();
```

Comment:
An exception out of the cctor will result in failure to load the type completely. I dont think you want to prevent other users from being unable to consume EventSource just because the allocation had an exception that went unhandled in cctor.

Either move this check to instance ctor or prevent the exception from remaining unhandled.